tuxedo-core/src/executive.rs

```diff
@@ -360,14 +389,85 @@ impl<
             block_hash
         );
 
-        // TODO, we need a good way to map our UtxoError into the supposedly generic InvalidTransaction
-        // https://paritytech.github.io/substrate/master/sp_runtime/transaction_validity/enum.InvalidTransaction.html
-        // For now, I just make them all custom zero
-        let r = Self::validate_tuxedo_transaction(&tx);
+        // Inherents are not permitted in the pool. They only come from the block author.
+        // We perform this check here rather than in the `validate_tuxedo_transaction` helper,
+        // because that helper is called again during on-chain execution. Inherents are valid
+        // during execution, so we do not want this check repeated.
+        let r = if tx.checker.is_inherent() {
+            Err(TransactionValidityError::Invalid(InvalidTransaction::Call))
+        } else {
+            // TODO, we need a good way to map our UtxoError into the supposedly generic InvalidTransaction
+            // https://paritytech.github.io/substrate/master/sp_runtime/transaction_validity/enum.InvalidTransaction.html
+            // For now, I just make them all custom zero
+            Self::validate_tuxedo_transaction(&tx)
+                .map_err(|_| TransactionValidityError::Invalid(InvalidTransaction::Custom(0)))
+        };
 
         debug!(target: LOG_TARGET, "Validation result: {:?}", r);
 
-        r.map_err(|_| TransactionValidityError::Invalid(InvalidTransaction::Custom(0)))
+        r
+    }
+
+    // The last two are for the standard beginning-of-block inherent extrinsics.
+    pub fn inherent_extrinsics(data: sp_inherents::InherentData) -> Vec<<B as BlockT>::Extrinsic> {
+        debug!(
+            target: LOG_TARGET,
+            "Entering `inherent_extrinsics`."
+        );
+
+        // Extract the complete parent block from the inheret data
+        let parent: B = data
+            .get_data(&PARENT_INHERENT_IDENTIFIER)
+            .expect("1")
+            .expect("2");
+
+        // Extract the beginning-of-block inherents from the previous block.
```

Comment:
improved in 6409334ff59118cda4dc786271e5e6ece396bd23

Reply:
> So this is extracting the end of block inherents from the previous block as the beginning of block inherents for this block?

No, that's not what's happening. end-of-block inherents are not implemented yet and aren't required for parachain support. Anything related to end-of-block inherents will come in the future.

This logic is extracting all of the inherents from the previous block. But they will not go into the next block. Once the previous inherents are extracted, they are passed into the `create_extrinsics` runtime api, an off-chain runtime api that is used for creating this blocks inherents.

So why do we need the old inherents to create the new ones? Because we need to know what outputs it created so the next block's inherent can use them as inputs or peeks.